src/test/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/TransactionIdValidationInterceptorTest.java

```diff
@@ -0,0 +1,89 @@
+
+package uk.gov.companieshouse.confirmationstatementapi.interceptor.validation;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.web.servlet.HandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.TRANSACTION_ID_KEY;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionIdValidationInterceptorTest {
+
+    static Stream<String> blankStrings() {
+        return Stream.of("", "   ", null);
+    }
+
+    @Mock
+    private HttpServletRequest mockHttpServletRequest;
+    @InjectMocks
+    private TransactionIdValidationInterceptor transactionIdValidationInterceptor;
+
+    @Test
+    void preHandleTrueForStringWithSpecialChars() throws Exception {
+        MockHttpServletResponse mockHttpServletResponse = new MockHttpServletResponse();
+        Object mockHandler = new Object();
+
+        String special = "1123-abc$456%def";
+        var pathParams = new HashMap<String, String>();
+        pathParams.put(TRANSACTION_ID_KEY, special);
+        when(mockHttpServletRequest.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE)).thenReturn(pathParams);
+
+        assertTrue(transactionIdValidationInterceptor.preHandle(mockHttpServletRequest, mockHttpServletResponse, mockHandler));
+    }
+
+    @Test
+    void preHandleTrueForFiftyLengthString() throws Exception {
+        MockHttpServletResponse mockHttpServletResponse = new MockHttpServletResponse();
+        Object mockHandler = new Object();
+
+        String fifty = "12345678901234567890123456789012345678901234567890";
+        var pathParams = new HashMap<String, String>();
+        pathParams.put(TRANSACTION_ID_KEY, fifty);
+        when(mockHttpServletRequest.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE)).thenReturn(pathParams);
+
+        assertTrue(transactionIdValidationInterceptor.preHandle(mockHttpServletRequest, mockHttpServletResponse, mockHandler));
+    }
+
+    @ParameterizedTest
+    @MethodSource("blankStrings")
+    void preHandleFalseForBlankString() {
```

Comment:
the test method needs a parameter for the different input params, something like preHandleFalseForBlankString(String input)
at the moment it runs 3 times but the transactionId is hardcoded to " " each time so need to set the input as transactionId like
        pathParams.put(TRANSACTION_ID_KEY, input);